src/ott/datasets.py

```diff
@@ -105,6 +110,168 @@ def _create_sample_generators(self) -> Iterator[jnp.array]:
       yield samples
 
 
+@dataclasses.dataclass
+class SklearnDistribution:
+  """A class to define toy probability 2-dimensional distributions.
+
+  Args:
+    name: the name specifying the centers of the mixture components:
+
+      - ``moon`` - lower moon Sklearn dataset,
+      - ``s_curve`` - s curve Sklearn dataset.
+      - ``swiss`` - swiss roll Sklearn dataset
+
+    init_rng: initial PRNG key
+    dim_data: data dimensionality
+    theta_rotation: angle to define the rotation matrix to rotate samples
+    offset: offset added to the Sklearn samples
+    scale: scaling factor to scale the Sklearn samples
+    std_noise: standard deviation of the gaussian additive noise
+    batch_size: batch size of the samples
+  """
+
+  name: Literal["moon", "s_curve", "swiss"]
+  init_rng: jax.Array
+  dim_data: int = 2
+  theta_rotation: float = 0.0
+  offset: Optional[jnp.ndarray] = None
+  scale: float = 1.0
+  std_noise: float = 0.01
+  batch_size: int = 1024
+
+  def __iter__(self) -> Iterator[jnp.ndarray]:
+    """Random sample generator from a Sklearn distribution.
+
+    Returns:
+    A generator of samples from the Sklearn distribution.
+    """
+    return self._create_sample_generators()
+
+  def __post_init__(self):
+
+    # check dimension consistency with dataset
+    if self.dim_data > 2:
+      assert self.name != "moon", (
+          "Moon disribution only supported in dimension 2."
+      )
+      assert self.theta_rotation == 0., (
+          "Samples rotation only supported in dimension 2."
+      )
+
+    # define offset
+    self._offset = jnp.zeros(
+        self.dim_data
+    ) if self.offset is None else self.offset
+
+    # define rotation matrix to rotate samples
+    if self.theta_rotation == 0.:
+      self.rotation_fn = lambda x: x
+    else:
+      rotation_marix = jnp.array([
+          [jnp.cos(self.theta_rotation), -jnp.sin(self.theta_rotation)],
+          [jnp.sin(self.theta_rotation),
+           jnp.cos(self.theta_rotation)],
+      ])
+      self.rotation_fn = lambda x: jnp.matmul(rotation_marix, x)
+
+  def _create_sample_generators(self) -> Iterator[jnp.ndarray]:
+    rng = jax.random.PRNGKey(0) if self.init_rng is None else self.init_rng
+
+    while True:
+      rng, _ = jax.random.split(rng)
```

Comment:
Please do `rng, rng_seed = jax.random.split(rng, 2)` and later use the `rng_seed`.